### DISCUSSION TITLE
Fabric on MacOS installation instructions here

### DISCUSSION THREAD

**Discussion by smrki1** [danielmiessler/Fabric#127] [General]
I have struggled for a week unsuccessfully to make Fabric work. Found this link, followed instructions and now it works. Hope it helps.
https://cpatocybersecurity.beehiiv.com/p/fabric

**Comment by CPAtoCybersecurity** [comment]
Glad it helped and thanks for sharing!

**Comment by danielmiessler** [comment]
This should not be necessary anymore after the new install process, but let us know.

**Reply by brannos** [reply]
I am still having issues installing on Mac Osx Catalina. Now have Python3.12.3 but still have unresolved . 'pipx install .' gives this:

user1@192-168-1-108 fabric % pipx install .      
'fabric' already seems to be installed. Not modifying existing installation in
'/Users/user1/.local/pipx/venvs/fabric'. Pass '--force' to force installation.

Tried the --force option but still got dependency conflicts as below:

Installing to existing venv 'fabric'
Fatal error from pip prevented installation. Full pip output in file:
    /Users/user1/.local/pipx/logs/cmd_2024-05-12_18.33.48_pip_errors.log

pip seemed to fail to build package:
    crewai-tools<0.2.0,>=0.1.1

Some possibly relevant errors from pip install:
    ERROR: Cannot install crewai-tools because these package versions have conflicting dependencies.
    ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts

⚠️  Not removing existing venv /Users/user1/.local/pipx/venvs/fabric because
    it was not created in this session
Error installing fabric from spec '/Users/user1/Desktop/AI/fabric'.

**Comment by smrki1** [comment]
Seems it is already installed, but you have a conflict with different versions. As stated in the output you got. Pretty easy to copy/paste this and ask chatgpt to help you with this.

**Comment by brannos** [comment]
ok, will try this. If know about deoendencies but I could not understand the suggestions on how to resolve. The article I read suggesting editing 2 config files to resolve and then suggested some other vague actions. It may have been that I used an older Mac which I have used Anacondo on. Acanconda installed some versions of software but often hung so it may have left me with differen versions of libraries, so I deinstalled Python 2.7 (comes with osx Catalina) and then installed Python 3.12.x but still no luck.

Will try Chatgpt idea first. Failing that I might do a complete reinstall of Catalina or forat the drive and go from scratch (after doing a time machine backup).

**Comment by brannos** [comment]
ok, I ditched Anaconda as it never finished loading. However have removed default 2.x verson of python (not the SYSTEM version). and reinstalling python 3.12. Still seems to have issues with the PATH in zsh.

I had these issues 20 years ago about multiple path localtions and it seems to have gotten WORST!!  About 10+ locations i saw on Stackoverflow (trusted I think). IS there a solution as fabric command failes and so did conda. so I think I should revert to bash

**Comment by brannos** [comment]
tried reverting to bash, deinstalling and reinstalling everything even both python versions and no luck. python 3.12.3 now installed bu still get pip seemed to fail to build package:
crewai-tools<0.2.0,>=0.1.1

Some possibly relevant errors from pip install:
ERROR: Cannot install crewai-tools because these package versions have conflicting dependencies.
ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts

and then fabric command not available.  

Can I send some config files for someone to look at ?  Is there a utility that can package these things for diagnosis?

**Reply by devonartis** [reply]
I had the same problem using zsh but when I run the command in bash shell I dont have the problem which shell are you using
